com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java

```diff
@@ -86,7 +87,9 @@
             options.put(Constants.PROJECTNAME, attachArguments.projectName);
         }
         ISourceLookUpProvider sourceProvider = context.getProvider(ISourceLookUpProvider.class);
-        sourceProvider.initialize(context.getDebugSession(), options);
+        sourceProvider.initialize(context, options);
+        IHotCodeReplaceProvider hcrProvider = context.getProvider(IHotCodeReplaceProvider.class);
```

Comment:
The provider should be initialized whenever the `DebugAdapterContext` is ready. Here's the flow of provider:
1. A provider is only called by handlers, but not the opposite
2. A provider can be initialized right after the `DebugAdapterContext` is ready, and it can save the reference for later use.
3. Whenever there is a request from the handlers, the provider can then use the context.

Reply:
Same as above